src/zm_logger.cpp

```diff
@@ -515,8 +515,9 @@ void Logger::logPrint( bool hex, const char * const filepath, const int line, co
     va_list     argPtr;
     struct timeval  timeVal;
 
-    const char * const file = basename(filepath);
-    
+    char *filecopy = strdup(filepath);
```

Comment:
Well, as I'm not programmer myself, I read manuals. It was said that such casting is a bad habit as const is a pointer to read only memory location and char* to read write. Recommendation was to use another pointer.